cmd/oras/internal/errors/errors.go

```diff
@@ -61,11 +60,16 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 }
 
 // NewErrEmptyTagOrDigest creates a new error based on the reference string.
-func NewErrEmptyTagOrDigest(ref registry.Reference) error {
-	return NewErrEmptyTagOrDigestStr(ref.String())
-}
-
-// NewErrEmptyTagOrDigestStr creates a new error based on the reference string.
-func NewErrEmptyTagOrDigestStr(ref string) error {
-	return fmt.Errorf("%q: no tag or digest when expecting <name:tag|name@digest>", ref)
+func NewErrEmptyTagOrDigest(ref string, cmd *cobra.Command, needsTag bool) error {
+	form := `"<name>@<digest>"`
+	errMsg := `no digest specified`
+	if needsTag {
+		form = fmt.Sprintf(`"<name>:<tag>" or %s`, form)
+		errMsg = "no tag or digest specified"
+	}
+	return &Error{
+		Err:            fmt.Errorf(`"%s": %s`, ref, errMsg),
+		Usage:          fmt.Sprintf("%s %s", cmd.Parent().CommandPath(), cmd.Use),
+		Recommendation: fmt.Sprintf(`You need to specify a reference in the form of %s. Run "%s -h" for more options and examples`, form, cmd.CommandPath()),
```

Comment:
Suggest avoiding second person pronoun to make the statement more formal.
```suggestion
		Recommendation: fmt.Sprintf(`Please specify a reference in the form of %s. Run "%s -h" for more options and examples`, form, cmd.CommandPath()),
```